pkg/externalauthproviders/flags.go

```diff
@@ -0,0 +1,295 @@
+			Required: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed("console-client-id") {
+		result.consoleClientId, err = interactive.GetString(interactive.Input{
+			Question: "Console client id",
+			Default:  result.consoleClientId,
+			Help:     cmd.Lookup("console-client-id").Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed("console-client-secret") {
+		if result.consoleClientId != "" {
+			// skips if no consoleClientId is provided
+			result.consoleClientSecret, err = interactive.GetString(interactive.Input{
+				Question: "Console client secret",
+				Default:  result.consoleClientSecret,
+				Help:     cmd.Lookup("console-client-secret").Usage,
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return result, nil
+}
+
+func CreateExternalAuthConfig(args *ExternalAuthProvidersArgs) (*cmv1.ExternalAuth, error) {
+	externalAuthBuilder := cmv1.NewExternalAuth().ID(args.name)
+
+	if args.issuerUrl != "" || args.issuerCaFile != "" || args.issuerAudiences != nil {
+		tokenIssuerBuilder := cmv1.NewTokenIssuer()
+
+		if args.issuerUrl != "" {
+			tokenIssuerBuilder.URL(args.issuerUrl)
+		}
+
+		if args.issuerCaFile != "" {
+			// Get certificate contents
+			ca := ""
+			if args.issuerCaFile != "" {
+				cert, err := os.ReadFile(args.issuerCaFile)
+				if err != nil {
+					return &cmv1.ExternalAuth{}, fmt.Errorf("expected a valid certificate bundle: %s", err)
+				}
+				ca = string(cert)
+			}
+			// Set the CA file, if any
+			if ca != "" {
+				tokenIssuerBuilder.CA(ca)
+			}
+		}
+
+		if args.issuerAudiences != nil {
+			tokenIssuerBuilder.Audiences(args.issuerAudiences...)
+		}
+		externalAuthBuilder.Issuer(tokenIssuerBuilder)
+	}
+
+	if args.claimMappingGroupsClaim != "" || args.claimMappingUsernameClaim != "" {
+		groupClaimBuilder := cmv1.NewGroupsClaim().Claim(args.claimMappingGroupsClaim)
+		usernameClaimBuilder := cmv1.NewUsernameClaim().Claim(args.claimMappingUsernameClaim)
+
+		// there's a bug here that username is not created in the backend. Not sure if it's a bug
+		tokenClaimMappingsBuilder := cmv1.NewTokenClaimMappings().Groups(groupClaimBuilder).
+			UserName(usernameClaimBuilder)
+		claimBuilder := cmv1.NewExternalAuthClaim().Mappings(tokenClaimMappingsBuilder)
+		externalAuthBuilder.Claim(claimBuilder)
+	}
+
+	if args.consoleClientId != "" || args.consoleClientSecret != "" {
+		clientBuilder := cmv1.NewExternalAuthClientConfig().
+			ID(args.consoleClientId).Secret(args.consoleClientSecret).Component(
+			// Component will be "fixed" with a "constant" component for the openshift console
+			cmv1.NewClientComponent().Name("console").Namespace("openshift-console"))
+		externalAuthBuilder.Clients(clientBuilder)
+	}
+
+	externalAuthConfig, err := externalAuthBuilder.Build()
+
+	// print does show up the username
+	fmt.Print(externalAuthConfig.Claim().Mappings().GetUserName())
```

Comment:
@lucasponce Not sure why, but the create does not successfully create two fields.
The first field is UserName claim. From line 271 to line 275, I tried to add the userNameClaim builder and I can print it in line 289. However, I checked the backend and got nil from that field. Not sure what causes it.

Same with client-secret. Client id does show up while secret didn't. Not sure if that's hidden on purpose from ocm get.

Reply:
With today's progress, we now also support the CA certificate create + pure command line creation without interactive mode. Aside from the two identified bugs, the remaining todo is the validation rules. I will work on it tomorrow. Target date is still by the end of the week and it's on track.

Reply:
Update - added the claim validation rules. Tomorrow with do a more thorough tests. Note, the ";" is not treated properly in input string. Changed to ":" and it works well now. @lucasponce to confirm if it's okay to use ":" instead.

```
--claim-validation-rule=<claim1>:<required_value1> \
--claim-validation-rule=<claim2>:<required_value2> \
--claim-validation-rule=<claim3>:<required_value3> \
```

Reply:
@andreadecorte and I identified that there was a mismatch for the casing, introducing json will fix it: https://github.com/openshift-online/ocm-api-model/pull/912

Reply:
Actually I wonder if we still need as it already supports multiple cases. Will wait until next week for Lucas' feedback

```
# Also Corba supports to add multiple attributes and parse them into an array

--claim-validation-rule=<claim1>;<required_value1> \
--claim-validation-rule=<claim2>;<required_value2> \
--claim-validation-rule=<claim3>;<required_value3> \

```